app/javascript/admin/controllers/confirmation_modal_controller.js

```diff
@@ -0,0 +1,101 @@
+import { Controller } from 'stimulus';
+import { adminModal } from '../adminModal';
+import { displaySnackbar } from '../displaySnackbar';
+import { displayErrorAlert } from '../displayErrorAlert';
+
+const confirmationText = (username) =>
+  `My username is @${username} and this action is 100% safe and appropriate.`;
+
+// const errorAlert = (errMsg) =>
+//   `<div class="crayons-notice crayons-notice--danger mb-3">${errMsg}</div>`;
+
+export default class ConfirmationModalController extends Controller {
+  static targets = [
+    'confirmationModalAnchor',
+    'confirmationTextField',
+    'confirmationTextWarning',
+  ];
+
+  closeConfirmationModal() {
+    this.confirmationModalAnchorTarget.innerHTML = '';
+    document.body.style.height = 'inherit';
+    document.body.style.overflowY = 'inherit';
+  }
+
+  removeBadgeAchievement(id) {
+    return document.querySelector(`[data-row-id="${id}"]`).remove();
+  }
+
+  async sendToEndpoint({ itemId, endpoint }) {
+    try {
+      const response = await fetch(`${endpoint}/${itemId}`, {
+        method: 'DELETE',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+          'X-CSRF-Token': document.querySelector("meta[name='csrf-token']")
+            ?.content,
+        },
+        credentials: 'same-origin',
+      });
+
+      const outcome = await response.json();
+
+      if (response.ok) {
+        this.removeBadgeAchievement(itemId);
+        displaySnackbar(outcome.message);
+      } else {
+        displayErrorAlert(outcome.error);
+      }
+
+      this.closeConfirmationModal();
+    } catch (err) {
+      displayErrorAlert(err.message);
+    }
+  }
+
+  checkConfirmationText(event) {
+    const { itemId, endpoint, username } = event.target.dataset;
+
+    if (this.confirmationTextFieldTarget.value == confirmationText(username)) {
+      this.closeConfirmationModal();
+      this.sendToEndpoint({ itemId, endpoint });
+    } else {
+      this.confirmationTextWarningTarget.classList.remove('hidden');
+    }
+  }
+
+  confirmationModalBody(username) {
+    return `
+      <div class="crayons-field">
+        <p>To confirm this update, type in the sentence: <br />
+        <strong>${confirmationText(username)}</strong></p>
+        <div data-confirmation-modal-target="confirmationTextWarning" class="crayons-notice crayons-notice--warning hidden" aria-live="polite">
+          The confirmation text does not match.
+        </div>
+        <input
+          type="text"
+          data-confirmation-modal-target="confirmationTextField"
+          class="crayons-textfield flex-1 mr-2"
+          placeholder: "Confirmation text">
```

Comment:
```suggestion
        <input
          aria-label="Type the sentence above to confirm this update"
          aria-describedby="<ID of the p tag wrapping the 'to confirm this update...' text >"
          type="text"
          data-confirmation-modal-target="confirmationTextField"
          class="crayons-textfield flex-1 mr-2"
          placeholder: "Confirmation text">
```

It's quite an odd thing to try to describe to someone without the visual context... I can't quite decide what the best way to handle this is (tbh now I'm playing about with it I think it's not a very user-friendly way to confirm updates for someone who can't easily see the confirmation text with all its casing and punctuation, but that's a rabbithole we don't need to go down here 😅 ).

It's a bit verbose, but I think we should add an aria-label, and also link the instruction text using `aria-describedby`. That way, when a screen reader user goes to the input, they know what the field is for and also have the full instructions announced.